src/type/post.ts

```diff
@@ -0,0 +1,19 @@
+import { User } from './user';
+import { Comment } from './comment';
+
+export interface Post {
+  userId: number;
+  id: number;
+  title:string;
+  body:string;
+  user: User | null;
+  comments: Comment[] | null;
```

Comment:
Consider initializing 'comments' property with an empty array instead of null, for better type consistency.